app/views/products/show.html.erb

```diff
@@ -0,0 +1,93 @@
+<article>
+  <section class="product large-6 small-8 columns">
+    <h4 id="product-title"><%= @product.name %></h4>
+    <%= link_to @product.user.username, user_products_path(id: @product.user.id), class: "small" %><br />
+
+    <span class="filled-star"><%= "★ " * @average_rating %></span>
+    <span class="unfilled-star"><%= "★ " * (5 - @average_rating) %></span>
+    <a href="#reviews" class="small">
+      <% if @reviews.empty? %>
+        <%= "Be the first to review this item" %>
+      <% else %>
+        <%= "#{@reviews.count} customer reviews" %>
+      <% end %>
+    </a><br />
+    <%= image_tag(@product.image_url, class: "feature") %>
+  </section>
+
+  <section class="product-cart large-6 small-4 columns">
+    <ul>
+      <li>
+        <span class="small-gray">Price:</span> <span class="big-red"><%= number_to_currency(@product.price) %></span>
+      </li>
+      <% if @product.quantity <= 5 %>
+        <li>
+          <span class="big-red">Only <%= @product.quantity %> left in stock (more on the way). </span>
+          <span class="small">Ships from and sold by Amazon.com. Gift-wrap available.</span>
+        </li>
+      <% end %>
+    </ul>
+
+    <section class="add-to-cart">
+      <%= form_tag add_item_path,  method: :post do %>
+        <%= hidden_field_tag :id, @product.id %>
+        <%= label_tag :quantity, 'Qty:' %>
+        <%= select_tag :quantity, options_for_select(1..30) %><br />
+        <%= submit_tag "Add to cart" %>
+      <% end %>
+    </section>
+
+  </section>
+</article>
+
+<section class="product-details">
+  <h4>Product Details</h4>
+  <ul>
+    <li>
+      <strong>Description:</strong> <%= @product.description %>
+    </li>
+    <li>
+      <% if session[:user_id] == @product.user.id %>
+      <%= link_to "Edit Product", edit_product_path(@product.id) %>
+      <% end %>
+    </li>
+    <li>
+      <strong>Shipping Weight: </strong>4.2 ounces
+    </li>
+    <li>
+      <strong>Domestic Shipping: </strong>Item can be shipped within U.S.
+    </li>
+    <li>
+      <strong>International Shipping: </strong>This item can be shipped to select countries outside of the U.S.
+    </li>
+  </ul>
+</section>
+
+<section class="reviews">
+  <a name="reviews"></a>
+  <h4>Customer Reviews</h4>
+  <% if !@reviews.empty? %>
+    <span class="filled-star"><%= "★ " * @average_rating %></span>
+    <span class="unfilled-star"><%= "★ " * (5 - @average_rating) %></span>
```

Comment:
It would be good to have the unfilled logic on the model rather than here in the view